File: python-client/goredis/zsets.py
from .client import get_client

def ZAdd(key, *args):
    """Add one or more members to a sorted set, or update its score. args: score, member, ..."""
    return get_client().send_command("ZADD", key, *args)

def ZRem(key, *members):
    """Remove one or more members from a sorted set."""
    return get_client().send_command("ZREM", key, *members)

def ZScore(key, member):
    """Get the score associated with the given member in a sorted set."""
    return get_client().send_command("ZSCORE", key, member)

def ZCard(key):
    """Get the number of members in a sorted set."""
    return get_client().send_command("ZCARD", key)

def ZRange(key, start, stop, with_scores=False):
    """Return a range of members in a sorted set, by index."""
    args = ["ZRANGE", key, start, stop]
    if with_scores: args.append("WITHSCORES")
    return get_client().send_command(*args)

def ZRevRange(key, start, stop, with_scores=False):
    """Return a range of members in a sorted set, by index, with scores ordered high to low."""
    args = ["ZREVRANGE", key, start, stop]
    if with_scores: args.append("WITHSCORES")
    return get_client().send_command(*args)

def ZGet(key, member=None):
    """Get score of a member or all members with scores."""
    args = ["ZGET", key]
    if member: args.append(member)
    return get_client().send_command(*args)

File: go-client/zsets.go
package goredis

// ZAdd adds one or more members to a sorted set, or updates the score of existing members.
func ZAdd(key string, members ...interface{}) (interface{}, error) {
	cmdArgs := append([]interface{}{"ZADD", key}, members...)
	return mustGetClient().SendCommand(cmdArgs...)
}

// ZRem removes one or more members from a sorted set.
func ZRem(key string, members ...interface{}) (interface{}, error) {
	cmdArgs := append([]interface{}{"ZREM", key}, members...)
	return mustGetClient().SendCommand(cmdArgs...)
}

// ZScore retrieves the score of a member in a sorted set.
func ZScore(key string, member interface{}) (interface{}, error) {
	return mustGetClient().SendCommand("ZSCORE", key, member)
}

// ZCard returns the number of members in a sorted set.
func ZCard(key string) (interface{}, error) {
	return mustGetClient().SendCommand("ZCARD", key)
}

// ZRange retrieves members in a sorted set within the given range.
func ZRange(key string, start, stop int, withScores bool) (interface{}, error) {
	cmdArgs := []interface{}{"ZRANGE", key, start, stop}
	if withScores {
		cmdArgs = append(cmdArgs, "WITHSCORES")
	}
	return mustGetClient().SendCommand(cmdArgs...)
}

// ZRevRange retrieves members in a sorted set within the given range in reverse order.
func ZRevRange(key string, start, stop int, withScores bool) (interface{}, error) {
	cmdArgs := []interface{}{"ZREVRANGE", key, start, stop}
	if withScores {
		cmdArgs = append(cmdArgs, "WITHSCORES")
	}
	return mustGetClient().SendCommand(cmdArgs...)
}

// ZGet retrieves the member(s) and their scores from a sorted set.
func ZGet(key string, member ...interface{}) (interface{}, error) {
	cmdArgs := []interface{}{"ZGET", key}
	if len(member) > 0 {
		cmdArgs = append(cmdArgs, member[0])
	}
	return mustGetClient().SendCommand(cmdArgs...)
}
